coordinator/src/services/headers.ts

```diff
@@ -0,0 +1,36 @@
+import Hapi from "@hapi/hapi"
+import {isProd} from "./environment"
+import {fhir, validationErrors as errors} from "@models"
+
+export enum RequestHeaders {
+  REQUEST_ID = "nhsd-request-id",
+  AUTH_LEVEL = "nhsd-identity-authentication-level",
+  RAW_RESPONSE = "x-raw-response",
+  SKIP_VALIDATION = "x-skip-validation",
+  SMOKE_TEST = "x-smoke-test"
+}
+
+export const invalidProdHeaders: Array<RequestHeaders> = [RequestHeaders.RAW_RESPONSE, RequestHeaders.SKIP_VALIDATION]
+
+export const rejectInvalidProdHeaders: Hapi.Lifecycle.Method = (
+  request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit
+) => {
+  if (isProd()) {
+    const listOfInvalidHeaders = Object.keys(request.headers).filter(
+      requestHeader => invalidProdHeaders.includes(requestHeader as RequestHeaders)
+    )
+    if (listOfInvalidHeaders.length) {
+      console.error(`Request with id: ${
+        request.headers[RequestHeaders.REQUEST_ID]
+      } had invalid header(s): ${
+        listOfInvalidHeaders.length
```

Comment:
Would be nicer to stringify the invalid headers rather than just including the length, but not a big deal

Reply:
For future ref, you dont need to include the request ID, the logger will store that for you